specfile/changelog.py

```diff
@@ -0,0 +1,128 @@
+
+    Attributes:
+        timestamp: Timestamp of the entry.
+        author: Author of the entry.
+        evr: EVR (epoch, version, release) of the entry.
+        content: List of lines forming the content of the entry.
+        legacy: Use legacy (date only) timestamp format.
+    """
+
+    def __init__(
+        self,
+        timestamp: datetime.datetime,
+        author: str,
+        evr: str,
+        content: List[str],
+        legacy: bool = False,
+    ) -> None:
+        self.timestamp = timestamp
+        self.author = author
+        self.evr = evr
+        self.content = content.copy()
+        self.legacy = legacy
+
+    def __str__(self) -> str:
+        return f"{self.header}\n" + "\n".join(self.content) + "\n"
+
+    def __repr__(self) -> str:
+        timestamp = repr(self.timestamp)
+        content = repr(self.content)
+        legacy = repr(self.legacy)
+        return f"ChangelogEntry({timestamp}, '{self.author}', '{self.evr}', {content}, {legacy})"
+
+    @property
+    def header(self) -> str:
+        """Entry header in the format that is being used in Fedora spec files."""
+        a = arrow.Arrow.fromdatetime(self.timestamp)
+        if not self.legacy:
+            dt = a.format("ddd MMM DD hh:mm:ss ZZZ YYYY")
+        else:
+            dt = a.format("ddd MMM DD YYYY")
+        return f"* {dt} {self.author} - {self.evr}"
+
+
+class Changelog(collections.UserList):
+    """
+    Class that represents a changelog.
+
+    Attributes:
+        data: List of individual entries.
+    """
+
+    def __str__(self) -> str:
+        return "\n".join(str(i) for i in reversed(self.data))
+
+    def __repr__(self) -> str:
+        data = repr(self.data)
+        return f"Changelog({data})"
+
+    @staticmethod
+    def parse(section: Section) -> "Changelog":
+        """
+        Parses a %changelog section.
+
+        Args:
+            section: Section to parse.
+
+        Returns:
+            Constructed instance of `Changelog` class.
+        """
+        header_regex = re.compile(
+            r"^\* \w+ (\w+ \d{2} (?:\d{2}:\d{2}:\d{2} [A-Z]+ )?\d{4}) (.+) - (.+)$"
+        )
+        data: List[ChangelogEntry] = []
+        for line in section:
+            m = header_regex.match(line)
+            if m:
+                dt, author, evr = m.groups()
+                try:
+                    datetime = arrow.get(dt, "MMM DD hh:mm:ss ZZZ YYYY").datetime
+                except arrow.parser.ParserError:
+                    try:
+                        datetime = arrow.get(dt, "MMM DD YYYY").datetime
+                    except arrow.parser.ParserError as e:
+                        raise SpecfileException(
+                            "Failed to parse timestamp of changelog entry"
+                        ) from e
+                    else:
+                        legacy = True
+                else:
+                    legacy = False
+                data.insert(0, ChangelogEntry(datetime, author, evr, [], legacy))
```

Comment:
Would it be possible to use keyword arguments to make this explicit? (I am speaking about the empty list.)
```suggestion
                data.insert(0, ChangelogEntry(datetime, author, evr, content=[], legacy=legacy))
```